miden-tx/src/compiler/mod.rs

```diff
@@ -73,7 +73,12 @@ impl TransactionCompiler {
     ) -> Result<AccountCode, TransactionCompilerError> {
         let account_code = AccountCode::new(account_code, &self.assembler)
             .map_err(TransactionCompilerError::LoadAccountFailed)?;
-        self.account_procedures.insert(account_id, account_code.procedures().to_vec());
+
+        self.account_procedures.insert(
+            account_id,
+            account_code.procedures().iter().map(|(p, _)| *p).collect::<Vec<Digest>>(),
```

Comment:
I would probably add an `AccountCode::procedure_roots()` method so that things like this could be just `account_code.procedure_roots().collect::<Vec<_>>()`.

Reply:
Here would we want to return an `iterator` over a new `Vec<Digest>` or would we want to return a `Vec<Digest>` by how you wrote the comment it seems that you would want an `iterator` returned, that's the method I followed for now let me know if it needs change.